server-spi-private/src/main/java/org/keycloak/provider/EnvironmentDependentProviderFactory.java

```diff
@@ -31,7 +31,10 @@ public interface EnvironmentDependentProviderFactory {
      * @return <code>true</code> if the provider is supported and should be available, <code>false</code> otherwise
      * @deprecated Prefer overriding/using the {@link #isSupported(Config.Scope)} method.
      */
-    boolean isSupported();
+    @Deprecated
+    default boolean isSupported() {
+        return false;
+    }
```

Comment:
I wouldn't change it. When some factory implements this interface, the assumption behind it is that there's really a need to check the environment setup and enable the factory based on specific conditions. I'd require to override the method.

I understand your input as both methods should be overridden when the Config.Scope is needed. However, AFAIK, there are only 2 cases (out of >70) when the scope is needed, otherwise the plain `isSupported()` is used. 
The JavaDoc is IMHO little bit misleading as I wouldn't consider it as a method, that should be deprecated. I think it should only emphasize the preference with the `isSupported(Config.Scope ...)` over the no-args method. 

@shawkins Does it work for you?

Reply:
@mabartos I think this is not about which use cases need `Config.Scope` and which don't. The interface is defined in a way that `isSupported()` is deprecated and shouldn't be use in new implementations. Problem is, you must use it as it has no default value.

Reply:
Here's what I'm comfortable doing in this pr (as it's a small change):

- remove the deprecation
- change the deprecated method to have a default implementation

Or I can back the change out and we can log an issue for:

- removing the legacy method, which touches quite a few classes and I assume would hit the community as a breaking change as well
- introduce a new interface with the desired signature, so that the old interface can be deprecated - this will lead to a proper deprecation path where users can switch to the new interface to resolve the deprecation prior to the old interface being removed.

Reply:
Removing the deprecation seems a bit out of scope of this PR given it's used at many places. I'd do that as a separate follow-up (we'll also need to involve the core team for that). For now we can just add a default impl for the deprecated method.

@mabartos Would that work for you?

Reply:
Yep, removing the deprecation requires explicit approval from the core team.  

> removing the legacy method, which touches quite a few classes and I assume would hit the community as a breaking change as well

Even when the deprecation is present there for some time, removing the method will not bring any additional value, quite the opposite. IMHO, the UX would be worse for the custom impls as I can imagine for the majority of cases, the scope is not needed at all - introduce such breaking change does not seem right to me. But it's for a different discussion. 

> @mabartos Would that work for you?

@vmuzikar @shawkins Yes, I prefer to add the default impl for the no-args method.

Reply:
> Removing the deprecation seems a bit out of scope of this PR given it's used at many places. I'd do that as a separate follow-up (we'll also need to involve the core team for that). For now we can just add a default impl for the deprecated method.

To clarify I mean that we should do one of these, not both. If we leave the method with a default impl and deprecated for a major release, then I'm fine with it being removed later.

But if you're thinking there doesn't seem to be any need to every remove it, then I'm also fine with opening another issue on removing the deprecation.

Reply:
I think we definitely should remove the deprecated method at some point, just maybe not in this PR. So I'm ok with just adding the default impl to it for now.

Reply:
Correct, I'm not suggesting removing the deprecated method in this pr. 

The only consideration I'm raising about that could be in the scope of this pr is removing the deprecation - that is leave the method but not as deprecated and just assume we won't get around to ever removing it.

Reply:
> The only consideration I'm raising about that could be in the scope of this pr is removing the deprecation - that is leave the method but not as deprecated and just assume we won't get around to ever removing it.

Seems it was deprecated for a reason, not sure we should simply un-deprecate it. :)

Reply:
I can understand the intent behind why it was deprecated - but unfortunately that isn't the right way to do it. Without the default impl, it still needs to be implemented. Then even if deprecated you generally won't know it as a developer - the IDE will typically only show deprecated usage (instance.isSupported() - which here is not directly called by the implementor), nor will it be by default in the compile deprecation warnings when you run maven / javac. 

So the deprecation isn't really helpful. The only way to make that work as expected would be to introduce another interface and deprecate the old interface.